### PR TITLE
Add Java syntax highlighting

### PR DESCRIPTION
**Pull request by milto-github**

### CODE REVIEW

**Comment by milto-github** [comment]
The file should be up to date with your most recent code. I'm having issues updating my repo to match yours but I'll figure it out

**Comment by matthewrsj** [comment]
@mndrenika this has conflicts with my master branch that need to be resolved before I can merge. It also has the commits from your previous PR. If you create a branch with only the commit(s) that you want included in this PR and create a PR from that branch the issues should be resolved.

File: kilo.c [hunk 1]
```diff
@@ -167,6 +167,19 @@ char *PY_HL_keywords[] = {
...
+char *J_HL_extensions[] = {".java",NULL};
+char *J_HL_keywords[] = {
+        "abstract","assert","break","case","catch","class","const","continue",
+        "default","do","else","enum","extends","final","finally","for","goto",
+        "if","implements","import","instanceof","interface","native","new",
+        "package","private","protected","public","return","static","strictfp",
+        "super","switch","synchronized","this","throw","throws","transient",
+        "try","void","volatile","while",
+        /* Java types */
+        "boolean","byte","char","double","float","int","long","short",NULL
```
Comment (matthewrsj) [inline kilo.c]: For the java types, add a `|` character after the text, like this: `"boolean|"`. The pipe signifies kilo to make that word a different color. Without the pipe char the type syntax color won't be different than the keywords, so `if` and `int` would be the same color.

Comment (milto-github) [reply kilo.c]: will do.

**Comment by milto-github** [comment]
Are the conflicts resolved now? Added the pipes as you said

**Comment by matthewrsj** [comment]
Hey @mndrenika the content looks good, but I'm looking at 7 commits, two of them are identical and have been merged in the previous PR, two of them are merge commits, and three of them have very nearly the same commit message. 

Could you create a new branch (other than master, which has your previous work on it) and add your changes to that branch? You should only need one commit on that branch to introduce the changes. If you want you can cherry pick the commits over to the new branch and squash the commits into one.

When that is done, you can open a PR from that branch to my master branch. The PR, and the history, will be much cleaner.

Here is how I would do it:
1. Set upstream remote `git remote add upstream https://github.com/matthewrsj/kilo.git`
2. `git fetch upstream`
3. `git merge upstream/master`
4. `git checkout -b java-syntax`
5. apply your update (you can use cherry picking or manually add it again)
6. Make a single commit with description of changes
7. `git push origin java-syntax`
8. Open a PR from java-syntax to my master branch

I'm not sure what your git experience is, so here are some references:
https://git-scm.com/docs/git-cherry-pick
https://git-scm.com/book/ch5-2.html
https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

**Pull request closed without merging**